engine/src/main/java/org/terasology/utilities/modifiable/ModifiableValue.java

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.utilities.modifiable;
+
+import org.terasology.module.sandbox.API;
+
+/**
+ * A helper type to get and modify the value of a component without changing its actual value.
+ * <p>
+ * Components using this type must mention so in their javadoc so all modifiers are added correctly.
+ * </p>
+ */
+@API
+public class ModifiableValue {
+    private final float baseValue;
+
+    private float preModifiers;
+    private float multipliers;
+    private float postModifiers;
```

Comment:
As these are simple float values they should not be named in plural form (that makes more sense for lists or arrays):
```suggestion
    private float preModifier;
    private float multiplier;
    private float postModifier;
```